cmd/collector/app/metrics.go

```diff
@@ -136,19 +166,43 @@ func newMetricsBySvc(factory metrics.Factory, category string) metricsBySvc {
 	spansFactory := factory.Namespace(metrics.NSOptions{Name: "spans", Tags: nil})
 	tracesFactory := factory.Namespace(metrics.NSOptions{Name: "traces", Tags: nil})
 	return metricsBySvc{
-		spans:  newCountsBySvc(spansFactory, category, maxServiceNames),
-		traces: newCountsBySvc(tracesFactory, category, maxServiceNames),
+		spans:  newSpanCountsBySvc(spansFactory, category, maxServiceNames),
+		traces: newTraceCountsBySvc(tracesFactory, category, maxServiceNames),
 	}
 }
 
-func newCountsBySvc(factory metrics.Factory, category string, maxServiceNames int) countsBySvc {
-	return countsBySvc{
-		counts: map[string]metrics.Counter{
-			otherServices: factory.Counter(metrics.Options{Name: category, Tags: map[string]string{"svc": otherServices, "debug": "false"}}),
-		},
-		debugCounts: map[string]metrics.Counter{
-			otherServices: factory.Counter(metrics.Options{Name: category, Tags: map[string]string{"svc": otherServices, "debug": "true"}}),
+func newTraceCountsBySvc(factory metrics.Factory, category string, maxServices int) traceCountsBySvc {
+	return traceCountsBySvc{
+		counts:      newTraceCountsOtherServices(factory, category, "false"),
+		debugCounts: newTraceCountsOtherServices(factory, category, "true"),
+		factory:     factory,
+		lock:        &sync.Mutex{},
+		maxServices: maxServices + numOfSamplerTypes, // numOfSamplerType is the offset added to maxServices threshold
```

Comment:
multiply?

Reply:
oh, about this, I thought if we keep using the 4000 threshold for `serviceName * numOfSamplerType`, the `numberOfSamplerTypes` added here will just be "compensating" for extra slots occupied by `otherServices[SamplerType]`